wandb/internal/file_stream.py

```diff
@@ -73,19 +75,25 @@ class CRDedupeFilePolicy(DefaultFilePolicy):
     """
 
     def process_chunks(self, chunks):
-        content = []
-        for line in [c.data for c in chunks]:
-            if content and content[-1].endswith('\r'):
-                content[-1] = line
-            else:
-                content.append(line)
+        ret = []
+        flag = False
+        for c in chunks:
+            lines = c.data.split(os.linesep)
+            for line in lines:
+                line = line.split('\r')[-1]
+                if line:
+                    if line.endswith('\x1b\x5b\x41'):
+                        if flag:
+                            ret.pop()
+                            flag = False
+                    else:
+                        ret.append(line + os.linesep)
+                        flag = True
```

Comment:
What does `flag` actually mean here?  Can we name this variable more descriptively.  A comment above the `endswith` escape sequence would be helpful too.